src/browser/BrowserService.cpp

```diff
@@ -507,6 +493,34 @@ void BrowserService::removeStoredPermissions()
     }
 }
 
+QList<Entry*> BrowserService::sortEntries(QList<Entry*>& pwEntries, const QString& host, const QString& entryUrl)
```

Comment:
Why don't you just use a QMultiMap from the start instead of copying from and to a QList? That way you could also just sort the list with a custom comparator.